src/bootstrap.coffee

```diff
@@ -32,20 +31,21 @@ bootstrap = ->
 		userConfig.deviceType ?= 'raspberry-pi'
 		if userConfig.registered_at?
 			return userConfig
-		deviceRegister.register(resinApi, userConfig)
-		.catch DuplicateUuidError, ->
-			resinApi.get
-				resource: 'device'
-				options:
-					filter:
-						uuid: userConfig.uuid
-				customOptions:
-					apikey: userConfig.apiKey
-			.then ([ device ]) ->
-				return device
-		.then (device) ->
+
+		deviceRegister.register(
+			userId: userConfig.userId
+			applicationId: userConfig.applicationId
+			uuid: userConfig.uuid
+			deviceType: userConfig.deviceType
+			deviceApiKey: userConfig.deviceApiKey
+			provisioningApiKey: userConfig.apiKey
+			apiEndpoint: config.apiEndpoint
+		)
+		.then ({ id }) ->
 			userConfig.registered_at = Date.now()
-			userConfig.deviceId = device.id
+			userConfig.deviceId = id
+			# Delete the provisioning key now.
+			delete userConfig.apiKey
```

Comment:
Keeping the provisioning key around would destroy the point of this PR :P  The device key can have permissions extended to create additional devices that depend on it though, although that'd be part of the API PR